### PR TITLE
Sparsity netx pr

### PR DESCRIPTION
**Pull request by Michaeljurado42**
Issue Number:  #239

Objective of pull request: This code provides netX with the ability to interpret synapses as sparse types when converting SNNs from lava-dl to lava. 

## Pull request checklist

Your PR fulfills the following requirements:
- [x] [Issue](https://github.com/lava-nc/lava-dl/issues) created that explains the change and why it's needed
- [x] Tests are part of the PR (for bug fixes / features)
- [x] [Docs](https://github.com/lava-nc/docs) reviewed and added / updated if needed (for bug fixes / features)
- [x] PR conforms to [Coding Conventions](https://lava-nc.org/developer_guide.html#coding-conventions)
- [x] [PR applys BSD 3-clause or LGPL2.1+ Licenses](https://lava-nc.org/developer_guide.html#add-a-license) to all code files
- [x] Lint (`flakeheaven lint src/lava tests/`) and (`bandit -r src/lava/.`) pass locally
- [x] Build tests (`pytest`) passes locally

## Pull request type

 

 

Please check your PR type:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation changes
- [ ] Other (please describe): 

## What is the current behavior?

-

## What is the new behavior?

-

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/lava/lib/dl/netx/hdf5.py [hunk 1]
```diff
@@ -53,6 +53,12 @@ class Network(AbstractProcess):
         neuron's reset parameter. None means no reset. Defaults to None.
     reset_offset: int
         determines the phase shift of network reset if enabled. Defaults to 0.
+    sparsity_map : boolean array, optional
```
Comment (bamsumit) [inline src/lava/lib/dl/netx/hdf5.py:56]: Let's use `sparse_fc_layer` instead of `sparsity_map`

**Review by bamsumit — COMMENTED** [review]
@Michaeljurado42 This looks good. Thanks for this one. Only one suggestion. Let's use `sparse_fc_layer` instead of `sparsity_map`

File: src/lava/lib/dl/netx/hdf5.py (hunk 1 above)
Comment (Michaeljurado42) [reply src/lava/lib/dl/netx/hdf5.py:56]: One problem with calling it` sparse_fc_layer` I could see is that sparsity may be applied intentionally to other layer types besides fully connected layers in the future.  For example, it is conceivable to reap some benefit from sparse convolutional operations on the Loihi2. This all depends on your (and intel's broadly) roadmap of development for lava and loihi2 though.

Comment (bamsumit) [reply src/lava/lib/dl/netx/hdf5.py:56]: Sparse convolution might be handled differently. That's what I meant `sparse_fc_layer` just toggles the sparsity of dense layer for now. At a later point `sparse_conv_layer` can similarly be introduced.

Comment (Michaeljurado42) [reply src/lava/lib/dl/netx/hdf5.py]: Okay it has been renamed. I also added some simple unit tests that check to see if the synapses are indeed sparse when `sparse_fc_layer`  or (in the case of blocks) `sparse_synapse` is set to True.

**Comment by Michaeljurado42** [comment]
Also, a final naming change which may be beneficial could be to rename the `Dense` proces defined in `netx.blocks` to `FullyConnected` or something. This would help us to not violate the naming convention in lava which differentiates between `dense` and `sparse` layers. Similarly `create_dense` would become `create_fully_connected` or something. Your previous comment brought this to my attention. 

Thank you for the feedback so far!

**Comment by bamsumit** [comment]
> Also, a final naming change which may be beneficial could be to rename the `Dense` proces defined in `netx.blocks` to `FullyConnected` or something. This would help us to not violate the naming convention in Lava which differentiates between `dense` and `sparse` layers. Similarly `create_dense` would become `create_fully_connected`. Your previous comment brought this to my attention.
> 
> Thank you for the feedback so far!

On board with that. It would break a lot of things at this stage of this project though.

**Review by bamsumit — APPROVED** [review]
Thanks a lot @Michaeljurado42

**Pull request merged**